### PR TITLE
don't continue with rt request if the token cache item doesn't contain RT

### PR DESCRIPTION
**Pull request by weijjia**
#695 Outlook is seeing from the Nullpointer exception during silent request. 
To prevent app from crashing, we don't continue the silent request if the token cache item doesn't contain RT. Also, when reading the JSON response from server, we mark the flag isMRRT as true only when server returns resource as well as RT in the JSON response. 

Add unit tests to verify that if no rt exists in the cache, we return the null result.

### CODE REVIEW

File: adal/src/main/java/com/microsoft/aad/adal/Oauth2.java [hunk 1]
```diff
@@ -255,7 +255,9 @@ public static AuthenticationResult processUIResponseParams(Map<String, String> r
                     expiresIn == null || expiresIn.isEmpty() ? AuthenticationConstants.DEFAULT_EXPIRATION_TIME_SEC
                             : Integer.parseInt(expiresIn));
 
-            if (response.containsKey(AuthenticationConstants.AAD.RESOURCE)) {
+            final String refreshToken = response.get(AuthenticationConstants.OAuth2.REFRESH_TOKEN);
```
Comment (addev-ashish) [inline adal/src/main/java/com/microsoft/aad/adal/Oauth2.java]: >refreshToken [](start = 25, length = 12)

you can use this refreshToken in line 288

**Comment by addev-ashish** [comment]
:shipit:

File: adal/src/main/java/com/microsoft/aad/adal/Oauth2.java (hunk 1 above)
Comment (weijjia) [inline adal/src/main/java/com/microsoft/aad/adal/Oauth2.java]: done.

**Comment by kpanwar** [comment]
:shipit: :rocket: :metal:
